daemon/oci_linux.go

```diff
@@ -816,27 +816,21 @@ func (daemon *Daemon) createSpec(c *container.Container) (retSpec *specs.Spec, e
 		return nil, err
 	}
 
-	var hasSecretDir bool
 	defer func() {
-		if hasSecretDir && err != nil {
+		if err != nil {
 			daemon.cleanupSecretDir(c)
+			daemon.cleanupConfigDir(c)
 		}
 	}()
 
-	if err := daemon.setupSecretDir(c, &hasSecretDir); err != nil {
+	if err := daemon.setupSecretDir(c); err != nil {
 		return nil, err
 	}
 
-	if err := daemon.setupConfigDir(c, &hasSecretDir); err != nil {
+	if err := daemon.setupConfigDir(c); err != nil {
```

Comment:
Could we further simplify this by having only a single tmpfs mount for both secrets and configs? I don't think keeping them separate is buying us much.

Reply:
Possibly, but they will still need to be namespaced due to potential name collisions, and then the mounting behavior changed.
For now it's probably easier to keep them separate and we could look at a refactor later.

Reply:
Looking a little closer, maybe it's not so big a change.

Reply:
Windows makes this a little more tricky.